packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/NamedTypeSymbolProvider.cs

```diff
@@ -375,6 +377,23 @@ private static string GetFullyQualifiedName(ITypeSymbol typeSymbol)
             // Handle generic types
             if (typeSymbol is INamedTypeSymbol namedTypeSymbol && namedTypeSymbol.IsGenericType)
             {
+                // handle nullable types
+                if (typeSymbol.NullableAnnotation == NullableAnnotation.Annotated && !IsCollectionType(namedTypeSymbol))
+                {
+                    const string nullableTypeName = "System.Nullable";
+                    var argTypeSymbol = namedTypeSymbol.TypeArguments.FirstOrDefault();
+
+                    if (argTypeSymbol != null)
+                    {
+                        if (argTypeSymbol.TypeKind == TypeKind.Error)
+                        {
+                            return GetFullyQualifiedName(argTypeSymbol);
+                        }
+
+                        string[] typeArguments = [.. namedTypeSymbol.TypeArguments.Select(GetFullyQualifiedName)];
+                        return $"{nullableTypeName}`{namedTypeSymbol.TypeArguments.Length}[{string.Join(", ", typeArguments)}]";
```

Comment:
Does the CSharpType we create from a symbol match the behavior for PropertyProvider.Type? I thought when creating CSharpType from the output types for a nullable, the CSharpType underlying Type would not be Nullable.

Reply:
Discussed offline and this shouldn't be an issue. I've added unit tests to validate this scenario.